nb2workflow/deploy.py

```diff
@@ -376,7 +410,47 @@ def _build_with_docker(git_origin,
             "last_change_time": meta['last_change_time'],
             "workflow_dispatcher_signature": workflow_dispatcher_signature,
             "workflow_nb_signature": workflow_nb_signature,
-            "dockerfile_content": dockerfile_content}
+            "dockerfile_content": dockerfile_content,
+            "resources": meta["resources"]}
+
+
+def append_k8s_secrets(resources, container, namespace):
+    secrets = []
+    for name, resource in resources.items():
+        verify_resource_secret(name, resource['required'], namespace=namespace)
+        for env in resource['env_vars']:
+            secrets.append(
+                {
+                    "name": env,
+                    "valueFrom": {
+                        "secretKeyRef": {"name": name, "key": "credentials"}
+                    }
+                }
+            )
+    if len(secrets) > 0:
+        container["env"] = secrets
+
+
+def get_k8s_secrets(namespace="oda-staging"):
+    json_data = sp.check_output(["kubectl", "get", "secrets", "-n", namespace, "-o", "json"])
+    items = json.loads(json_data)['items']
+    for secret in items:
+        yield secret['metadata']['name'], secret['data']
+
+
+def verify_resource_secret(name, required, namespace="oda-staging"):
+    # credentials
+    for secret_name, secret in get_k8s_secrets(namespace=namespace):
+        if secret_name == name:
+            if 'credentials' not in secret:
+                raise NameError(f"No credentials defined for secret {name}")
+            return True
+    message = f"No secrets defined for {name}"
+    if required:
+        raise FileNotFoundError(message)
```

Comment:
Exception type is misleading, secret is not a file

Reply:
Changed to RuntimeException